File: src/taptap_mcp/data/leaderboard_docs.py
"""TapTap 排行榜 API 文档"""

LEADERBOARD_DOCUMENTATION = {
    "title": "TapTap 排行榜系统",
    "description": "为小游戏提供完整的竞技排名和分数管理功能",
    "categories": {
        "score_submission": {
            "title": "分数提交",
            "description": "提交玩家分数到排行榜系统",
            "apis": [
                {
                    "name": "提交单个分数",
                    "method": "submitScore",
                    "description": "为当前玩家提交分数到指定排行榜",
                    "parameters": {
                        "leaderboardId": "string, 排行榜唯一标识",
                        "score": "number, 玩家分数",
                        "metadata": "object, 可选的元数据信息"
                    },
                    "example": """
// 提交分数到排行榜
const leaderboard = tap.getLeaderboardManager();

try {
    const result = await leaderboard.submitScore({
        leaderboardId: 'weekly_high_score',
        score: 15000,
        metadata: {
            level: 10,
            playTime: 120,
            achievements: ['perfect_game', 'speed_run']
        }
    });

    console.log('分数提交成功:', result.rank);
    console.log('当前排名:', result.position);

} catch (error) {
    console.error('分数提交失败:', error);
}
"""
                },
                {
                    "name": "批量提交分数",
                    "method": "submitScores",
                    "description": "一次性提交多个排行榜的分数",
                    "parameters": {
                        "scores": "Array<ScoreEntry>, 分数数组"
                    },
                    "example": """
// 批量提交多个分数
const scores = [
    {
        leaderboardId: 'daily_score',
        score: 8500,
        metadata: { mode: 'normal' }
    },
    {
        leaderboardId: 'weekly_score',
        score: 12000,
        metadata: { mode: 'hard' }
    }
];

try {
    const results = await leaderboard.submitScores(scores);
    results.forEach((result, index) => {
        console.log(`排行榜 ${scores[index].leaderboardId}:`, result.rank);
    });
} catch (error) {
    console.error('批量提交失败:', error);
}
"""
                }
            ]
        },
        "ranking_query": {
            "title": "排名查询",
            "description": "查询排行榜数据和玩家排名信息",
            "apis": [
                {
                    "name": "获取排行榜数据",
                    "method": "getLeaderboard",
                    "description": "获取指定排行榜的排名数据",
                    "parameters": {
                        "leaderboardId": "string, 排行榜ID",
                        "options": "object, 查询选项"
                    },
                    "example": """
// 获取排行榜前100名
const topPlayers = await leaderboard.getLeaderboard('global_ranking', {
    offset: 0,
    limit: 100,
    timeRange: 'all_time'  // 'daily', 'weekly', 'monthly', 'all_time'
});

console.log('排行榜数据:', topPlayers.entries);
topPlayers.entries.forEach((entry, index) => {
    console.log(`${index + 1}. ${entry.playerName}: ${entry.score}分`);
});
"""
                },
                {
                    "name": "获取玩家排名",
                    "method": "getPlayerRank",
                    "description": "获取当前玩家在指定排行榜中的排名",
                    "example": """
// 获取当前玩家排名
const playerRank = await leaderboard.getPlayerRank('global_ranking');

if (playerRank.hasRank) {
    console.log('当前排名:', playerRank.position);
    console.log('当前分数:', playerRank.score);
    console.log('超越玩家数:', playerRank.position - 1);
} else {
    console.log('玩家暂无排名记录');
}
"""
                },
                {
                    "name": "获取周围玩家",
                    "method": "getNearbyPlayers",
                    "description": "获取玩家排名附近的其他玩家",
                    "example": """
// 获取排名附近的玩家
const nearbyPlayers = await leaderboard.getNearbyPlayers('global_ranking', {
    range: 10  // 获取上下各10名玩家
});

console.log('附近玩家:', nearbyPlayers.entries);
console.log('我的位置:', nearbyPlayers.playerIndex);
"""
                }
            ]
        },
        "leaderboard_ui": {
            "title": "排行榜界面",
            "description": "显示和管理排行榜界面",
            "apis": [
                {
                    "name": "打开排行榜页面",
                    "method": "openLeaderboard",
                    "description": "打开 TapTap 内置的排行榜页面",
                    "parameters": {
                        "leaderboardId": "string, 可选，指定打开的排行榜"
                    },
                    "example": """
// 打开排行榜页面
try {
    await leaderboard.openLeaderboard('global_ranking');
} catch (error) {
    console.error('打开排行榜失败:', error);
}

// 打开排行榜列表页面（不指定具体排行榜）
await leaderboard.openLeaderboard();
"""
                }
            ]
        }
    },
    "integration_patterns": {
        "title": "集成模式",
        "description": "常用的排行榜集成模式和最佳实践",
        "patterns": [
            {
                "name": "游戏结束提交分数",
                "description": "在游戏结束时自动提交分数并显示排名变化",
                "example": """
class GameSession {
    constructor() {
        this.leaderboard = tap.getLeaderboardManager();
        this.currentScore = 0;
    }

    async endGame() {
        try {
            // 提交分数
            const result = await this.leaderboard.submitScore({
                leaderboardId: 'main_leaderboard',
                score: this.currentScore
            });

            // 显示排名信息
            this.showGameOverUI({
                score: this.currentScore,
                rank: result.position,
                isNewRecord: result.isPersonalBest
            });

        } catch (error) {
            console.error('提交分数失败:', error);
            // 显示游戏结束界面，但不显示排名
            this.showGameOverUI({
                score: this.currentScore,
                error: '排名暂时无法获取'
            });
        }
    }
}
"""
            },
            {
                "name": "实时排名显示",
                "description": "在游戏过程中实时显示玩家排名信息",
                "example": """
class RankingDisplay {
    constructor() {
        this.leaderboard = tap.getLeaderboardManager();
        this.updateInterval = null;
    }

    async startRealTimeUpdates() {
        // 初始加载
        await this.updateRanking();

        // 定期更新（避免过于频繁）
        this.updateInterval = setInterval(() => {
            this.updateRanking();
        }, 60000); // 每分钟更新一次
    }

    async updateRanking() {
        try {
            const playerRank = await this.leaderboard.getPlayerRank('main_leaderboard');
            const nearbyPlayers = await this.leaderboard.getNearbyPlayers('main_leaderboard', {
                range: 3
            });

            this.displayRankingInfo(playerRank, nearbyPlayers);
        } catch (error) {
            console.error('更新排名失败:', error);
        }
    }

    stopRealTimeUpdates() {
        if (this.updateInterval) {
            clearInterval(this.updateInterval);
        }
    }
}
"""
            }
        ]
    },
    "best_practices": {
        "title": "最佳实践",
        "description": "排行榜开发的最佳实践和建议",
        "practices": [
            {
                "category": "性能优化",
                "items": [
                    "避免频繁查询排行榜数据，建议使用缓存",
                    "使用分页加载来处理大量排行榜数据",
                    "批量提交分数以减少 API 调用次数",
                    "在合适的时机更新排名，避免过度刷新"
                ]
            },
            {
                "category": "用户体验",
                "items": [
                    "在分数提交时显示加载状态",
                    "优雅处理网络错误和超时",
                    "为排名变化提供视觉反馈",
                    "提供离线模式的本地排行榜"
                ]
            },
            {
                "category": "数据管理",
                "items": [
                    "合理设计排行榜的时间周期（日榜、周榜、月榜）",
                    "为不同难度或模式创建独立排行榜",
                    "记录足够的元数据用于反作弊检测",
                    "定期清理过期的排行榜数据"
                ]
            }
        ]
    }
}

LEADERBOARD_SEARCH_INDEX = {
    "keywords": {
        "排行榜": ["score_submission", "ranking_query", "leaderboard_ui"],
        "分数": ["score_submission", "ranking_query"],
        "排名": ["ranking_query", "leaderboard_ui"],
        "提交": ["score_submission"],
        "查询": ["ranking_query"],
        "界面": ["leaderboard_ui"],
        "UI": ["leaderboard_ui"],
        "实时": ["integration_patterns"],
        "性能": ["best_practices"],
        "优化": ["best_practices"]
    }
}

File: src/data/leaderboardDocs.ts
/**
 * TapTap Minigame Leaderboard API Documentation
 * Based on: https://developer.taptap.cn/minigameapidoc/dev/api/open-api/leaderboard/
 */

export interface LeaderboardAPI {
  name: string;
  method: string;
  description: string;
  parameters?: Record<string, string>;
  returnValue?: string;
  example: string;
}

export interface LeaderboardCategory {
  title: string;
  description: string;
  apis: LeaderboardAPI[];
}

export interface LeaderboardDocumentation {
  title: string;
  description: string;
  apiReference: string;
  categories: Record<string, LeaderboardCategory>;
}

export const LEADERBOARD_DOCUMENTATION: LeaderboardDocumentation = {
  title: "TapTap Minigame Leaderboard API",
  description: `Complete leaderboard functionality for TapTap minigames, including score submission, ranking queries, and leaderboard display.

⚠️ IMPORTANT:
- NO npm packages or SDK installation required
- NO imports needed
- The 'tap' object is a GLOBAL object provided by TapTap runtime environment
- All APIs are accessed via: tap.getLeaderboardManager()
- Works ONLY in TapTap minigame environment (not in regular web browsers)
- All methods accept a SINGLE object parameter (not multiple parameters)`,
  apiReference: "https://developer.taptap.cn/minigameapidoc/dev/api/open-api/leaderboard/",

  categories: {
    initialization: {
      title: "Initialization",
      description: "Get the LeaderboardManager instance from the global 'tap' object (provided by TapTap runtime)",
      apis: [
        {
          name: "tap.getLeaderboardManager",
          method: "tap.getLeaderboardManager()",
          description: "Get the LeaderboardManager instance to access leaderboard functionality. ⚠️ IMPORTANT: 'tap' is a GLOBAL object provided by TapTap runtime, NO imports or installations needed.",
          returnValue: "LeaderboardManager - The leaderboard manager instance",
          example: `// ⚠️ IMPORTANT: 'tap' is a global object, NO imports needed!
// This works ONLY in TapTap minigame environment

// Get LeaderboardManager instance
const leaderboardManager = tap.getLeaderboardManager();

// Now you can use leaderboardManager to call various methods
// All methods accept a SINGLE object parameter
leaderboardManager.openLeaderboard({
  leaderboardId: "your_id",
  collection: "public",
  callback: { onSuccess: (res) => {}, onFailure: (code, msg) => {} }
});`
        }
      ]
    },

    display: {
      title: "Display Leaderboard",
      description: "Open and display the leaderboard UI page",
      apis: [
        {
          name: "openLeaderboard",
          method: "leaderboardManager.openLeaderboard({ leaderboardId, collection, callback })",
          description: "Opens TapTap's leaderboard page, displaying total and friend leaderboards for the specified leaderboard ID",
          parameters: {
            "leaderboardId": "string (required) - Unique identifier for the leaderboard",
            "collection": "string (optional) - Leaderboard type: 'friends' for friend rankings or 'public' (default) for global rankings",
            "callback.onSuccess": "function (optional) - Success callback function",
            "callback.onFailure": "function (optional) - Failure callback with (code, message) parameters"
          },
          returnValue: "void - Opens native leaderboard UI",
          example: `// Open leaderboard UI
const leaderboardManager = tap.getLeaderboardManager();

// Open global leaderboard
leaderboardManager.openLeaderboard({
  leaderboardId: "weekly_high_score",
  collection: "public",
  callback: {
    onSuccess: function(res) {
      console.log("Leaderboard opened successfully:", res);
    },
    onFailure: function(code, message) {
      console.error(\`Failed to open leaderboard: code=\${code}, message=\${message}\`);
      // Error codes:
      // 500001: Leaderboard ID not found
      // 1025: Friend relationship permissions not declared
      // 104/103: User privacy/authorization issues
    }
  }
});

// Open friend leaderboard
leaderboardManager.openLeaderboard({
  leaderboardId: "friend_ranking",
  collection: "friends",
  callback: {
    onSuccess: function(res) {
      console.log("Friend leaderboard opened");
    },
    onFailure: function(code, message) {
      console.error("Failed:", message);
    }
  }
});`
        }
      ]
    },

    score_submission: {
      title: "Score Submission",
      description: "Submit player scores to leaderboards",
      apis: [
        {
          name: "submitScores",
          method: "leaderboardManager.submitScores({ scores, callback })",
          description: "Batch submit user leaderboard scores. Maximum of 5 scores can be submitted at once.",
          parameters: {
            "scores": "Array<ScoreEntry> (required) - Array of score entries to submit, maximum 5 entries",
            "scores[].leaderboardId": "string (required) - Unique identifier for the leaderboard",
            "scores[].score": "number (required) - Integer score value to submit",
            "callback.onSuccess": "function (optional) - Success callback function",
            "callback.onFailure": "function (optional) - Failure callback with (code, message) parameters"
          },
          returnValue: "void - Submission result is returned via callback",
          example: `// Submit scores to multiple leaderboards
const leaderboardManager = tap.getLeaderboardManager();

leaderboardManager.submitScores({
  scores: [
    {
      leaderboardId: "leaderboard_1",
      score: 1000
    },
    {
      leaderboardId: "leaderboard_2",
      score: 2000
    }
  ],
  callback: {
    onSuccess: function(res) {
      console.log("submitScores success:", res);
    },
    onFailure: function(code, message) {
      console.error(\`submitScores failed: \${code}, \${message}\`);
      // Error codes:
      // 500001: Leaderboard ID not found
      // 500002: Leaderboard parameter error
      // 500199: Invalid number of score entries (exceeds 5)
    }
  }
});

// Submit single score
leaderboardManager.submitScores({
  scores: [
    {
      leaderboardId: "weekly_high_score",
      score: 15000
    }
  ],
  callback: {
    onSuccess: function(res) {
      console.log("Score submitted successfully");
    },
    onFailure: function(code, message) {
      console.error("Failed to submit score:", message);
    }
  }
});`
        }
      ]
    },

    score_query: {
      title: "Score Query",
      description: "Query leaderboard scores and rankings",
      apis: [
        {
          name: "loadLeaderboardScores",
          method: "leaderboardManager.loadLeaderboardScores({ leaderboardId, collection, maxSize, nextPage, periodToken, callback })",
          description: "Retrieve paginated leaderboard data with support for friend and public rankings",
          parameters: {
            "leaderboardId": "string (required) - Unique identifier for the leaderboard",
            "collection": "string (optional) - Leaderboard type: 'friends' or 'public' (default)",
            "maxSize": "number (required) - Limit results between 1-200 entries",
            "nextPage": "string | undefined (optional) - Token for pagination to get next page, pass undefined for first request",
            "periodToken": "string | undefined (optional) - Time period identifier for the leaderboard",
            "callback": "CommonCallback (optional) - Callback object with onSuccess and onFailure",
            "callback.onSuccess": "function (optional) - Success callback function",
            "callback.onFailure": "function (optional) - Failure callback with (code, message) parameters"
          },
          returnValue: "void - Leaderboard data is returned via callback including scores list and pagination details",
          example: `// Load top scores from a leaderboard
const leaderboardManager = tap.getLeaderboardManager();

leaderboardManager.loadLeaderboardScores({
  leaderboardId: "your_leaderboardId",
  collection: "friends",
  maxSize: 10,
  nextPage: undefined,  // undefined for first request
  periodToken: undefined,
  callback: {
    onSuccess: function(res) {
      console.log("Leaderboard scores:", res);
      // res contains:
      // - leaderboard info
      // - scores list
      // - nextPage token for pagination
      // - isTruncated flag
    },
    onFailure: function(code, message) {
      console.error("Load failed:", message);
    }
  }
});

// Load global leaderboard with pagination
let nextPageToken = undefined;

function loadNextPage() {
  leaderboardManager.loadLeaderboardScores({
    leaderboardId: "weekly_high_score",
    collection: "public",
    maxSize: 50,
    nextPage: nextPageToken,
    periodToken: undefined,
    callback: {
      onSuccess: function(res) {
        console.log("Page loaded:", res.scores.length, "entries");

        // Check if there are more pages
        if (res.isTruncated) {
          nextPageToken = res.nextPage;
          // Can load next page with the new token
        }
      },
      onFailure: function(code, message) {
        console.error("Failed to load page:", message);
      }
    }
  });
}`
        },
        {
          name: "loadCurrentPlayerLeaderboardScore",
          method: "leaderboardManager.loadCurrentPlayerLeaderboardScore({ leaderboardId, collection, periodToken, callback })",
          description: "Get the current user's score and ranking position in the specified leaderboard. Requires user to have submitted a score to this leaderboard.",
          parameters: {
            "leaderboardId": "string (required) - Unique identifier for the leaderboard",
            "collection": "string (optional) - Leaderboard type: 'friends' or 'public' (default)",
            "periodToken": "string (optional) - Time period identifier for the leaderboard",
            "callback.onSuccess": "function (optional) - Success callback function",
            "callback.onFailure": "function (optional) - Failure callback with (code, message) parameters"
          },
          returnValue: "void - Current player's score data is returned via callback",
          example: `// Get current player's ranking
const leaderboardManager = tap.getLeaderboardManager();

leaderboardManager.loadCurrentPlayerLeaderboardScore({
  leaderboardId: "weekly_high_score",
  collection: "public",
  periodToken: undefined,
  callback: {
    onSuccess: function(res) {
      console.log("Your score:", res.score);
      console.log("Your rank:", res.rank);
      console.log("Leaderboard:", res.leaderboard);
    },
    onFailure: function(code, message) {
      console.error("Failed to load player score:", message);
    }
  }
});

// Get friend leaderboard ranking
leaderboardManager.loadCurrentPlayerLeaderboardScore({
  leaderboardId: "friend_ranking",
  collection: "friends",
  periodToken: undefined,
  callback: {
    onSuccess: function(res) {
      console.log("Your rank among friends:", res.rank);
    },
    onFailure: function(code, message) {
      console.error("Failed:", message);
    }
  }
});`
        },
        {
          name: "loadPlayerCenteredScores",
          method: "leaderboardManager.loadPlayerCenteredScores({ leaderBoardId, collection, maxCount, periodToken, callback })",
          description: "Retrieve scores for the current user and nearby players on a leaderboard, useful for showing surrounding competitors. ⚠️ NOTE: Parameter name is 'leaderBoardId' (capital B), not 'leaderboardId'.",
          parameters: {
            "leaderBoardId": "string (required) - Unique identifier for the leaderboard (⚠️ Note: capital 'B')",
            "collection": "string (required) - Leaderboard type: 'friends' or 'public'",
            "maxCount": "number (required) - Limit results between 1-25 players",
            "periodToken": "string (required) - Time period identifier for the leaderboard",
            "callback": "CommonCallback (required) - Callback object with onSuccess and onFailure",
            "callback.onSuccess": "function (optional) - Success callback function",
            "callback.onFailure": "function (optional) - Failure callback with (code, message) parameters"
          },
          returnValue: "void - Nearby players' scores are returned via callback",
          example: `// Load nearby players' scores
const leaderboardManager = tap.getLeaderboardManager();

leaderboardManager.loadPlayerCenteredScores({
  leaderBoardId: "your_leaderboardId",  // ⚠️ Note: capital 'B'
  collection: "friends",
  maxCount: 10,
  periodToken: "",
  callback: {
    onSuccess: function(res) {
      console.log("Players around you:", res);
      // res contains current player and nearby players' scores
      res.scores.forEach(score => {
        const marker = score.isCurrentPlayer ? '👉' : '  ';
        console.log(\`\${marker} Rank #\${score.rank}: \${score.playerName} - \${score.score}\`);
      });
    },
    onFailure: function(code, message) {
      console.error("Failed to load nearby scores:", message);
    }
  }
});

// Load surrounding players on global leaderboard
leaderboardManager.loadPlayerCenteredScores({
  leaderBoardId: "global_ranking",  // ⚠️ Note: capital 'B'
  collection: "public",
  maxCount: 20,
  periodToken: "",
  callback: {
    onSuccess: function(res) {
      console.log("Loaded", res.scores.length, "players around you");
    },
    onFailure: function(code, message) {
      console.error("Failed:", message);
    }
  }
});`
        }
      ]
    },

    common_scenarios: {
      title: "Common Implementation Scenarios",
      description: "Complete examples for typical use cases",
      apis: [
        {
          name: "Complete Game Flow",
          method: "N/A",
          description: "Example of integrating leaderboard into a complete game flow",
          example: `// Complete leaderboard integration example
const leaderboardManager = tap.getLeaderboardManager();

// 1. After game ends, submit score
async function submitGameScore(finalScore) {
  try {
    // Use leaderboardId (not leaderboardName!)
    // Wrap in scores array within an object
    await leaderboardManager.submitScores({
      scores: [{
        leaderboardId: 'daily_ranking',  // Use leaderboardId
        score: finalScore
      }]
    });
    console.log('Score submitted!');
    return true;
  } catch (error) {
    console.error('Submit failed:', error);
    return false;
  }
}

// 2. Show player's ranking after submission
async function showPlayerRanking() {
  try {
    // Use object parameter with leaderboardId
    const playerScore = await leaderboardManager.loadCurrentPlayerLeaderboardScore({
      leaderboardId: 'daily_ranking',
      collection: 'public'
    });

    // Display ranking to player
    showMessage(\`Your rank: #\${playerScore.rank}\`);
    showMessage(\`Your score: \${playerScore.score}\`);

    // Show if player improved
    if (playerScore.previousRank && playerScore.rank < playerScore.previousRank) {
      showMessage(\`🎉 You moved up \${playerScore.previousRank - playerScore.rank} positions!\`);
    }
  } catch (error) {
    console.error('Failed to get ranking:', error);
  }
}

// 3. Display nearby competitors
async function showNearbyPlayers() {
  try {
    // Use object parameter with leaderboardId
    const nearby = await leaderboardManager.loadPlayerCenteredScores({
      leaderboardId: 'daily_ranking',
      before: 3,
      after: 3
    });

    // Render leaderboard UI
    renderLeaderboard(nearby.entries);
  } catch (error) {
    console.error('Failed to load nearby players:', error);
  }
}

// 4. Open full leaderboard when player clicks "View All"
function openFullLeaderboard() {
  // Use object parameter with leaderboardId
  leaderboardManager.openLeaderboard({
    leaderboardId: 'daily_ranking',
    collection: 'public'
  });
}

// Complete flow
async function handleGameEnd(finalScore) {
  const submitted = await submitGameScore(finalScore);
  if (submitted) {
    await showPlayerRanking();
    await showNearbyPlayers();
  }
}
`
        },
        {
          name: "Pagination Example",
          method: "N/A",
          description: "Example of implementing paginated leaderboard browsing",
          example: `// Paginated leaderboard implementation
class LeaderboardView {
  constructor() {
    this.leaderboardManager = tap.getLeaderboardManager();
    this.currentPage = 0;
    this.pageSize = 20;
    this.leaderboardName = 'global_ranking';
  }

  async loadPage(page) {
    try {
      const offset = page * this.pageSize;
      const data = await this.leaderboardManager.loadLeaderboardScores(
        this.leaderboardName,
        {
          offset: offset,
          limit: this.pageSize
        }
      );

      this.renderPage(data);
      this.currentPage = page;

      // Calculate total pages
      const totalPages = Math.ceil(data.total / this.pageSize);
      this.updatePagination(page, totalPages);

      return data;
    } catch (error) {
      console.error('Failed to load page:', error);
      throw error;
    }
  }

  async nextPage() {
    await this.loadPage(this.currentPage + 1);
  }

  async previousPage() {
    if (this.currentPage > 0) {
      await this.loadPage(this.currentPage - 1);
    }
  }

  renderPage(data) {
    // Render leaderboard entries
    const listElement = document.getElementById('leaderboard-list');
    listElement.innerHTML = data.entries.map(entry => \`
      <div class="leaderboard-entry">
        <span class="rank">#\${entry.rank}</span>
        <span class="player">\${entry.playerName}</span>
        <span class="score">\${entry.score}</span>
      </div>
    \`).join('');
  }

  updatePagination(current, total) {
    document.getElementById('page-info').textContent = \`Page \${current + 1} of \${total}\`;
    document.getElementById('prev-btn').disabled = current === 0;
    document.getElementById('next-btn').disabled = current === total - 1;
  }
}

// Usage
const leaderboardView = new LeaderboardView();
leaderboardView.loadPage(0);
`
        }
      ]
    }
  }
};

/**
 * Search leaderboard documentation by keyword
 */
export function searchLeaderboardDocs(query: string, category?: string): string[] {
  const results: string[] = [];
  const lowerQuery = query.toLowerCase();

  const categoriesToSearch = category
    ? [LEADERBOARD_DOCUMENTATION.categories[category]]
    : Object.values(LEADERBOARD_DOCUMENTATION.categories);

  for (const cat of categoriesToSearch) {
    if (!cat) continue;

    for (const api of cat.apis) {
      const searchText = `${api.name} ${api.description} ${api.example}`.toLowerCase();
      if (searchText.includes(lowerQuery)) {
        results.push(`
### ${api.name}

**Method:** \`${api.method}\`

**Description:** ${api.description}

${api.parameters ? `**Parameters:**
${Object.entries(api.parameters).map(([key, value]) => `- \`${key}\`: ${value}`).join('\n')}
` : ''}

${api.returnValue ? `**Returns:** ${api.returnValue}\n` : ''}

**Example:**
\`\`\`javascript
${api.example}
\`\`\`
`);
      }
    }
  }

  return results;
}

/**
 * Get overview of leaderboard system
 */
export function getLeaderboardOverview(): string {
  return `# ${LEADERBOARD_DOCUMENTATION.title}

${LEADERBOARD_DOCUMENTATION.description}

**Official API Reference:** ${LEADERBOARD_DOCUMENTATION.apiReference}

## Available Categories

${Object.entries(LEADERBOARD_DOCUMENTATION.categories).map(([key, cat]) => `
### ${cat.title}
${cat.description}

Available methods: ${cat.apis.map(api => `\`${api.name}\``).join(', ')}
`).join('\n')}

## Quick Start

⚠️ **CRITICAL: Use leaderboardId (NOT leaderboardName) and wrap parameters in objects!**

\`\`\`javascript
// 1. Get LeaderboardManager instance
const leaderboardManager = tap.getLeaderboardManager();

// 2. Submit a score (use leaderboardId and wrap in scores array)
await leaderboardManager.submitScores({
  scores: [{
    leaderboardId: 'my_leaderboard',  // Use leaderboardId (not leaderboardName)
    score: 1000
  }]
});

// 3. Query current player's rank (use object parameter)
const playerScore = await leaderboardManager.loadCurrentPlayerLeaderboardScore({
  leaderboardId: 'my_leaderboard',
  collection: 'public'
});
console.log('Your rank:', playerScore.rank);

// 4. Open leaderboard UI (use object parameter)
leaderboardManager.openLeaderboard({
  leaderboardId: 'my_leaderboard'
});
\`\`\`
`;
}

/**
 * Get detailed documentation for a specific category
 */
export function getCategoryDocs(category: string): string {
  const cat = LEADERBOARD_DOCUMENTATION.categories[category];
  if (!cat) {
    return `Category "${category}" not found. Available categories: ${Object.keys(LEADERBOARD_DOCUMENTATION.categories).join(', ')}`;
  }

  return `# ${cat.title}

${cat.description}

${cat.apis.map(api => `
## ${api.name}

**Method:** \`${api.method}\`

**Description:** ${api.description}

${api.parameters ? `### Parameters

${Object.entries(api.parameters).map(([key, value]) => `- **\`${key}\`**: ${value}`).join('\n')}
` : ''}

${api.returnValue ? `### Returns

${api.returnValue}
` : ''}

### Example

\`\`\`javascript
${api.example}
\`\`\`

---
`).join('\n')}
`;
}
